bootstrap/src/main.ts

```diff
@@ -44,6 +26,38 @@ try {
   console.error(error);
 }
 
+/* Minor code duplication of StartURLManager, but Typescript gives compile problems we can't easily ignore when we import it */
+function processApp2AppArgs(url?: string): void {
+  const queryString = url || location.search.substr(1);
+  let pluginId, windowManager;
+
+  queryString.split('&').forEach(part => {
+    const [key, value] = part.split('=').map(v => decodeURIComponent(v));
+    switch (key) {
+      case "pluginId":
+        (window as any)['GIZA_PLUGIN_TO_BE_LOADED'] = value;
```

Comment:
Now it is good time to eliminate all of those `(window as any)` because you touch all of them.
Let's augment `Window` interface like this 
interface Window {
  GIZA_PLUGIN_TO_BE_LOADED?: string;
  GIZA_SIMPLE_CONTAINER_REQUESTED?: boolean;
  GIZA_ENVIRONMENT?: string;
  ZOWE_SWM_SHOW_LOGIN?: boolean;
  // etc
}
You could add it to https://github.com/zowe/zlux-platform/blob/staging/interface/src/index.d.ts re-write all the usages to `window.GIZA_PLUGIN_TO_BE_LOADED` and so on.

Reply:
Yes